### ISSUE TITLE
Stop in gdiff

### ISSUE THREAD

**Issue by nhooyr** [airblade/vim-gitgutter#303]
Perhaps we should add the following to the readme or something to allow gitgutter to disable in diffs.

``` vim
 augroup diff
   autocmd!
   autocmd FilterWritePre * if &diff | GitGutterDisable | endif
   autocmd BufWinLeave fugitive://* if &diff | GitGutterEnable | endif
augroup END
```

Probably a better way to do this but that should work.

**Comment by lewicki-pk** [comment]
I was about to create a new issue because I just noticed that in my configuration.
I wonder if this can be disabled in such diff by default?

**Comment by airblade** [comment]
Sounds like a good idea.

According to the docs (`:help DiffOrig`), `diff` is a window-local option and it's possible to view the same buffer in diff mode in one window and "normally" in another window.

Vim-gitgutter only knows about buffers, not windows, so I'm not quite sure how to implement this.  Having said that, the code accounts for `&diff` when [jumping between hunks with `[c` and `]c`](https://github.com/airblade/vim-gitgutter/blob/78d83c7056e1973ed4cbf9b5b3f09263a3dbf968/plugin/gitgutter.vim#L128-L129).

Anyway, the autocommands in your example look asymmetric:

- `FilterWritePre` and `BufWinLeave` don't seem matched.
- `*` and `fugitive://*` aren't matched.

Any change we implement should handle looking at diffs using vim's built-in diff viewer, not just Fugitive.

**Comment by airblade** [comment]
Does this work for you?

```viml
augroup diff
  autocmd!
  autocmd FilterWritePre * if &diff | GitGutterDisable | endif
  autocmd FilterWritePost * if &diff | GitGutterEnable | endif
augroup END
```

**Comment by lewicki-pk** [comment]
This disables gitgutter only in the original window (git working copy) and extra marks (`+`/`-`/`~`) are still visible on the left window (git index version)- see attached screenshoot.

They are also in the wrong place (i.e. in the proper line, but in diff mode these lines are shifted).


![disabling_gitgutter](https://cloud.githubusercontent.com/assets/7014937/15372234/e10076b8-1d3f-11e6-8570-20d541ba28b7.png)

**Comment by somini** [comment]
This change was too aggressive for me.

If I trigger a `Gdiff` the gutter will be clear (which is OK), but when I close the Fugitive buffer and come back to the main buffer, the gutter stays disabled until a `FocusGained` or any buffer change.

If there a chance of putting this under a flag, enabled by default?

**Comment by airblade** [comment]
@somini Thanks for the feedback.  I've addressed it in 457f35e.
